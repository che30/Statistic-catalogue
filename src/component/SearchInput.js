import React, { useState } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import FetchData from '../container/Fetchdata';
import { SearchMovie } from '../actions';

const SearchInput = ({ getMovie }) => {
  const [movie, setMovie] = useState({
    movieTitle: '',
  });
  const handleChange = (e) => {
    if (e.target.id === 'input-title') {
      setMovie({
        ...movie, movieTitle: e.target.value,
      });
    }
  };
  const handleSubmit = () => {
    if (movie.movieTitle !== '') {
      FetchData(movie.movieTitle).then((res) => getMovie(res));
    }
    setMovie({
      movieTitle: '',
    });
  };
  return (
    <div className="d-flex align-items-baseline justify-content-around">
      <form className="d-flex mt-3">
        <input
          type="input"
          placeholder="enter movie title"
          id="input-title"
          value={movie.movieTitle}
          onChange={handleChange}
        />
        <button type="button" onClick={handleSubmit}>search </button>
      </form>
    </div>
  );
};
SearchInput.propTypes = {
  getMovie: PropTypes.func.isRequired,
};
const mapDispatchToProps = (dispatch) => ({
  getMovie: (movie) => dispatch(SearchMovie(movie)),
});
export default connect(null, mapDispatchToProps)(SearchInput);
