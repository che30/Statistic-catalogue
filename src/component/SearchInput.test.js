import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { createStore, compose, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';
import SearchInput from './SearchInput';
//  const initialState = {
//    title: 'batman',
//    loading: true,
//    movies: [],
//    error: '',
//    year: '',
//    imbdID: '',

//  };

describe('search input methods', () => {
  const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(rootReducer(), composeEnhancer(applyMiddleware(thunk)));
  test('should render the initial message before users search a movie', async () => {
    const { findByText } = render(
      <Provider store={store}>
        <SearchInput />
      </Provider>,
    );
    await findByText('search');
  });
  it('renders correctly', () => {
    const tree = renderer

      .create(<Provider store={store}>
        <SearchInput />
      </Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
