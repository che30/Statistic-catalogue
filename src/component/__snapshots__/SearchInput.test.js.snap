// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search input methods renders correctly 1`] = `
<div
  className=" search-form "
>
  <form
    className="d-flex w-100 justify-content-around text-center mt-3 "
  >
    <input
      className="w-25 input-form text-center"
      id="input-title"
      onChange={[Function]}
      placeholder="enter movie title"
      type="input"
      value=""
    />
    <div>
      <button
        className="bg-dark text-white px-3 rounded py-1"
        onClick={[Function]}
        type="button"
      >
        search 
      </button>
    </div>
  </form>
</div>
`;
